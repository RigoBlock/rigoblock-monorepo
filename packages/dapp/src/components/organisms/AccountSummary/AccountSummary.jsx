import './AccountSummary.scss'
import { BigNumber } from 'bignumber.js'
import { ETHTOWEI } from '../../../constants/utils'
import AccountItem, { ACCOUNT_ITEM_TYPES } from '../../molecules/AccountItem'
import Amount, { AMOUNT_SIZES } from '../../molecules/Amount'
import PropTypes from 'prop-types'
import React from 'react'

const AccountSummary = ({ provider, number, balance, tokenBalance }) => {
  const ethBalance = balance.div(ETHTOWEI)
  return (
    <div className="account-summary">
      <AccountItem
        appearance={ACCOUNT_ITEM_TYPES.FULL}
        provider={provider}
        number={number}
      />
      <div className="balances">
        <Amount value={ethBalance} symbol={'ETH'} size={AMOUNT_SIZES.SMALL} />
        <Amount value={tokenBalance} symbol={'GRG'} size={AMOUNT_SIZES.SMALL} />
      </div>
    </div>
  )
}

AccountSummary.propTypes = {
  provider: PropTypes.string.isRequired,
  balance: PropTypes.object,
  tokenBalance: PropTypes.object,
  number: PropTypes.string.isRequired
}

AccountSummary.defaultProps = {
  balance: new BigNumber(0),
  tokenBalance: new BigNumber(0)
}

export default AccountSummary
