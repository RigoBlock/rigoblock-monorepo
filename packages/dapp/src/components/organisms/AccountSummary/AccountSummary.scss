@import "../../_settings/variables.scss";

.account-summary {
  padding: $dist-m $dist-s $dist-s $dist-s;

  .account-item-full {
    .account-number {
      font-size: $small-font-size;
    }
  }

  .balances {
    width: 118px;
    margin-left: $dist-l;
    padding-top: $dist-s;

    > *:not(:first-child) {
      padding-top: $dist-2xs;
    }
  }
}

