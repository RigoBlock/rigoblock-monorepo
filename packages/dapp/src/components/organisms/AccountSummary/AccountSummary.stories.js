import '../../_settings/_base.scss'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import AccountSummary from './AccountSummary'
import React from 'react'

const props = {
  provider: 'metamask',
  number: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196',
  balance: '57999999999960203063'
}

storiesOf('Organisms/AccountSummary', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div
      style={{
        width: '335px',
        background: 'white',
        padding: '16px'
      }}
    >
      {story()}
    </div>
  ))
  .add('default', () => <AccountSummary {...props} />)
