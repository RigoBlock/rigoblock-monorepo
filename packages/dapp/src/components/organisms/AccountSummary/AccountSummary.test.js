import { shallow } from 'enzyme'
import AccountSummary from './AccountSummary'
import toJson from 'enzyme-to-json'

const props = {
  provider: 'metamask',
  number: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196',
  balance: '57999999999960203063'
}

describe('AccountSummary component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(AccountSummary, props)))
    ).toMatchSnapshot()
  })
})
