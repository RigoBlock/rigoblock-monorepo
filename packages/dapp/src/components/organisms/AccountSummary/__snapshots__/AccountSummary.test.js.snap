// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSummary component renders correctly 1`] = `
<div
  className="account-summary"
>
  <AccountItem
    appearance="full"
    number="0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196"
    provider="metamask"
  />
  <div
    className="balances"
  >
    <Amount
      size="small"
      symbol="ETH"
      symbolSize="small"
      value={"57.999999999960203063"}
      valueSize="small"
    />
    <Amount
      size="small"
      symbol="GRG"
      symbolSize="small"
      value={"0"}
      valueSize="small"
    />
  </div>
</div>
`;
